packages/@aws-cdk/aws-lambda/lib/alias.ts

```diff
@@ -193,6 +208,26 @@ export class Alias extends QualifiedFunctionBase implements IAlias {
     });
   }
 
+  /**
+   * Configure provisioned concurrency autoscaling on a function alias. Returns a scalable attribute that can call
+   * `scaleOnUtilization()` and `scaleOnSchedule()`.
+   *
+   * @param options Autoscaling options
+   */
+  public addAutoScaling(options: AutoScalingOptions): IScalableFunctionAttribute {
```

Comment:
Would this be a better user experience if we change this to a getter `autoscaling` that (instantiates, if not already, and) returns `scalableAlias`?

Reply:
I think we might want to keep the same pattern as we have in all other L2 which implements autoscaling (ecs, ddb, ec2)

Reply:
I agree with returning `IScalableFunctionAttribute` in the `addAutoScaling()` method. 

My question is if this should be -

```ts
export class ScalableFunctionAttribute extends appscaling.BaseScalableAttribute implements IScalableFunctionAttribute {
```

Keeps the type expectations in the correct place and any mismatches will be reported at the correct locations by the compiler.

Reply:
I see, you are right, I have change it!